src/bin/miri.rs

```diff
@@ -253,12 +233,15 @@ impl rustc_driver::Callbacks for MiriBeRustCompilerCalls {
     #[allow(rustc::potential_query_instability)] // rustc_codegen_ssa (where this code is copied from) also allows this lint
     fn config(&mut self, config: &mut Config) {
         if config.opts.prints.is_empty() && self.target_crate {
+            // Avoid warnings about unsupported crate types
+            #[allow(rustc::bad_opt_access)]
+            config.opts.crate_types.retain(|&c| c == CrateType::Executable || c == CrateType::Rlib);
```

Comment:
We should probably only do this for target crates?

What happens if the list of crate types is empty after the filtering?

Also, why is it okay to allow `bad_opt_access` here?